datanode/broker/sqlstore_broker.go

```diff
@@ -214,15 +220,15 @@ func (b *SQLStoreBroker) processBlock(ctx context.Context, dbContext context.Con
 			return nil, ctx.Err()
 		case err = <-errCh:
 			return nil, err
-		case <-slowTimeUpdateTicker.C:
+		case <-b.slowTimeUpdateTicker.C:
 			b.log.Warningf("slow time update detected, time between checks %v, block height: %d, total block processing time: %v", slowTimeUpdateThreshold,
 				block.Height, blockTimer.duration)
 		case e := <-eventsCh:
 			metrics.EventCounterInc(e.Type().String())
 			blockTimer.startTimer()
-			if e.Type() == events.TimeUpdate {
-				timeUpdate := e.(entities.TimeUpdateEvent)
 
+			switch e.Type() {
+			case events.EndBlockEvent:
```

Comment:
Can we have a check here to ensure that after end block event, the next event it a being block, anything else should cause a panic, reason is it will potentially result in inconsistent state across nodes (considering restore from snapshot)